### ISSUE TITLE
Global improvement of the API

### ISSUE THREAD

**Issue by Craftmusic-Git** [clapfoot/warapi#104]
Hi foxhole team !

I'm a developer, and my team of friends who play foxhole with me have a lot of cool features who can be added in the game api.
We decided to separate these features in 3 groups :

# Features

## Global features :

- Add statistics after the end of the war : (Nb of 120 mm shell crafted, the greatest killer, the greatest medic, ..., the most contested city ... )
- Add general clan statistic, ( name, nb of player, nb of active player this war ... ) , this could be fun

## Protection of intel data

- Protect your api with a steam OAuth account to access protected data. With this feature we can add a lot of "sensitive intel" of foxhole like : Stockpile : Private and Public, personnal statistic, clan private information, player location, factory, facilities ...
- Protect your API with an in-game generated password. (This can be a clan keyword)

## Some features but they require a secure authentication.

- Facilities or factory live datas. ( It can give a revival of the game, because super cool tools can be born with this feature )
- Stockpile liev datas
- Editing clan's configuration
- Player location
- Statistic during the war like number of items craft, kills ...

**Comment by Yin117** [comment]
Information such as player location and stockpiles are an awful idea due to alts, imagine if a major Coalition had an alt and made a Discord bot that tracked major stockpile changes such as a increase in 120mm shells...

I like the enthuasium, and some of these ideas, but some can become furthur exploits in my eyes.

**Comment by Craftmusic-Git** [comment]
I agree with the player location feature, but not for the private stockpiles. If we can see private stockpile with an api this can be a huge QoL upgrade.

**Comment by Pasqualle** [comment]
Generally when an API can verify the user, then the API should work with the user's access level.

But the availability of an information sometimes depends on other factors also. Like an existing facility is not even visible on the game map when not inside the intelligence radius.  Why would you expect any data from it.
The API should not expose anything which you do not see in the game. The expectation of the data returned from the API, is the information you can gather when you are standing in the home region. That means most of these features must be declined.

I would really like to see more external tools available for foxhole, but the data you would like to manage can't come from the API.
